### PR TITLE
[Source] Reddit: New Saved Post

### PR DESCRIPTION
**Pull request by vellames-turing**
Added new source that emit an event when user saves a post

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

**Comment by vellames-turing** [comment]
@dannyroosevelt  #1466 must be merged first to move on with this PR.

**Comment by dannyroosevelt** [comment]
> @dannyroosevelt #1865 must be merged first to move on with this PR.
> 
> For reviewer information: You just need to review `new-saved-post-by-user.mjs`. Other files already were reviewed in #1865

So to confirm, that is waiting on PR review, right?

**Comment by vellames-turing** [comment]
> > @dannyroosevelt #1865 must be merged first to move on with this PR.
> > For reviewer information: You just need to review `new-saved-post-by-user.mjs`. Other files already were reviewed in #1865
> 
> So to confirm, that is waiting on PR review, right?

It is ready for QA

**Comment by dannyroosevelt** [comment]
> > > @dannyroosevelt #1865 must be merged first to move on with this PR.
> > > For reviewer information: You just need to review `new-saved-post-by-user.mjs`. Other files already were reviewed in #1865
> > 
> > 
> > So to confirm, that is waiting on PR review, right?
> 
> It is ready for QA

Jacob QA'd all the other components, so I'm good to proceed.

**Comment by dannyroosevelt** [comment]
@vellames-turing it looks like there are some build errors to address, I think maybe you'll need to merge master in.

**Comment by vellames-turing** [comment]
> @vellames-turing it looks like there are some build errors to address, I think maybe you'll need to merge master in.

I need to wait for #1865 to proceed here :)

**Comment by vellames-turing** [comment]
#1865 merged.
We are good to review here

**Review by js07 — COMMENTED** [review]
@vellames-turing Looks good overall. I made a few minor comments and a suggestion about caching post IDs to avoid duplicating events.

File: components/reddit/reddit.app.mjs [hunk 1]
```diff
@@ -371,6 +371,28 @@ export default {
...
+      timeFilter,
+      includeSubredditDetails,
+      limit = 100,
+    ) {
+      const params = {
+        before,
+        show: "given",
+        sort: "new",
+        t: timeFilter,
+        type: "links",
```
[the remark this answers is not in the document]
Comment (vellames-turing) [reply components/reddit/reddit.app.mjs:386]: I don't think so, if I use type `comments` nothing is fetched.
Looks like use `type = links` or `type = null` don't make any difference in this context. I am removing it anyways

**Comment by andrewjschuang** [comment]
@js07 I believe @vellames-turing has addressed the comments above. I think this can be approved, right?

**Comment by js07** [comment]
> @js07 I believe @vellames-turing has addressed the comments above. I think this can be approved, right?

Yeah, it looks like the comments have been addressed.

If we want events to be emitted for posts that are re-saved, we could fetch and reset the cache with all saved posts on each run of the action.

**Comment by dannyroosevelt** [comment]
> > @js07 I believe @vellames-turing has addressed the comments above. I think this can be approved, right?
> 
> Yeah, it looks like the comments have been addressed.
> 
> If we want events to be emitted for posts that are re-saved, we could fetch and reset the cache with all saved posts on each run of the action.

I say let's publish! We can optimize / expand the feature-set later.

@vellames-turing plz merge when ready!

**Comment by dannyroosevelt** [comment]
Oh @andrewjschuang and @js07 I think you both have to approve first

**Review by andrewjschuang — APPROVED** [review]
Thanks for the contribution, @vellames-turing!

**Pull request merged**